### PR TITLE
Add command to delete SMS and Email logs for domain

### PR DESCRIPTION
**Pull request by MartinRiese**
## Product Description

Some domains have SMS logs referencing MessagingEvents and MessagingSubEvent from other domains. Since those logs will not be deleted by the `delete_domain` command an SQL exception is thrown for foreign key violations. To address this add a command that just delete the SMS and Email logs from a domain. So this command can be run first

## Technical Summary

* Add command to just delete SMS and Email logs
* Extract common command code for domain commands
* Add test to cover case with SMS logs on other domain

https://dimagi-dev.atlassian.net/browse/USH-3390

## Feature Flag

None

### Automated test coverage

corehq.apps.domain.tests.test_delete_domain:TestDeleteDomain.test_cross_over_messages

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [ ] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by esoergel — APPROVED** [review]

**Review by millerdev — CHANGES_REQUESTED** [review]
This is a harsh review, and I'm sorry for that. I have no ill will toward @MartinRiese or anyone else who worked on it. This is meant to be constructive feedback to hopefully ultimately improve CommCare HQ.

While this new command will unblock the current domain deletion effort, I don't think it's a good addition to the product. It feels more like a hacky workaround for a partially implemented feature than a holistic improvement.

**It is not discoverable.** It adds yet another management command to the mish-mash pile of random commands that do random things. You have to know about it and what problem it solves for it to be useful. If you were trying to delete a domain and got the `MessagingSubEvent` error, nothing would tell you to go use this command and then try deleting the domain again.

**It is not safe to run as a stand-alone operation.** It is only safe to be run just before ultimately running `delete_domain` on the same domains. I would prefer to see a solution where `delete_domain` can handle the scenario this was built for rather than adding yet another management command.

I think it is fine to use this branch in a private release to unblock the current domain deletion effort, but I'm not excited about having it merged as a permanent feature as-is.

**Comment by MartinRiese** [comment]
@millerdev No reason to apologize. I think this is a very valid concern.

I had thought about changing the existing command but deleting all sms/emails that reference a messaging subevent would require a inner select in PostgreSql as far as I can tell. That seemed rather expensive to add to the normal delete domain command.

Since the scenario with the cross domain references does not seem very common I think a private release would be a good way to go. Do you think just closing the PR afterwards and linking it in the story is appropriate? I think it would be good to preserve the intel gathered. It looks like GitHub keeps the diff even if you close a PR and delete the branch.

File: corehq/apps/domain/management/commands/delete_domain_base.py
[the remark this answers is not in the document]
Comment (millerdev) [reply corehq/apps/domain/management/commands/delete_domain_base.py:1]: Could put it there, or I would also consider putting the base class in `corehq/apps/domain/management/commands/delete_domain.py` and have other commands that need it import from there.

**Comment by millerdev** [comment]
> Do you think just closing the PR afterwards and linking it in the story is appropriate? I think it would be good to preserve the intel gathered. It looks like GitHub keeps the diff even if you close a PR and delete the branch.

Yes, I think that would be fine. We can always restore the branch later if the need arises.

**Pull request closed without merging**